compiler/compile/CompilationTypes.hpp

```diff
@@ -81,21 +81,22 @@ enum TR_CallingContext {
 
 
 
-// function return type flags for word preceding the entry point in any
+// function return types for word preceding the entry point in any
 // linkage that needs to be callable from interpreted.  Also used in pic
 // for use when dispatching methods that are interpreted in megamorphic send
 // case
 enum TR_ReturnInfo
    {
-   TR_VoidReturn        = 0,
-   TR_IntReturn         = 1,
-   TR_LongReturn        = 2,
-   TR_FloatReturn       = 3,
-   TR_DoubleReturn      = 4,
-   TR_ObjectReturn      = 5, // Currently 64-bit platforms only
-   TR_FloatXMMReturn    = 6, // IA32 and AMD64 only
-   TR_DoubleXMMReturn   = 7, // IA32 and AMD64 only
-   TR_ConstructorReturn = 8  // Return from constructor
+   TR_VoidReturn,
+   TR_ByteReturn,
+   TR_IntReturn,
+   TR_LongReturn,
+   TR_FloatReturn,
+   TR_DoubleReturn,
+   TR_ObjectReturn,     // Currently 64-bit platforms only
+   TR_FloatXMMReturn,   // IA32 and AMD64 only
+   TR_DoubleXMMReturn,  // IA32 and AMD64 only
+   TR_ConstructorReturn // Return from constructor
```

Comment:
I believe OpenJ9 encodes this value as part of the linkage info word in the code cache before the JITed entry point.  I don't think you can arbitrarily insert a new value in this enum without synchronizing the change with the VM in OpenJ9.  Please check this--maybe I'm wrong.

I realize this is an OpenJ9 concern but since the code for this lives in OMR (for now) it should be addressed here.

Reply:
I just realized it isn't a fair request on my part to ask you to delve into OpenJ9 to figure out whether or not this will break anything there since this PR is not being made in support of OpenJ9 work.  Sorry about that.  Given that, I'll ask that you move `TR_ByteReturn` to the end of this enum to minimize potential downstream churn and replace the explicit enumerator values that you deleted.